core/src/main/java/org/apache/any23/extractor/rdf/BaseRDFExtractor.java

```diff
@@ -105,7 +109,24 @@ public void run(
             parser.getParserConfig().addNonFatalError(BasicParserSettings.NORMALIZE_DATATYPE_VALUES);
             //ByteBuffer seems to represent incorrect content. Need to make sure it is the content
             //of the <script> node and not anything else!
-            parser.parse(in, extractionContext.getDocumentIRI().stringValue());
+            RDFFormat format = parser.getRDFFormat();
+            String iri = extractionContext.getDocumentIRI().stringValue();
+
+            if (format.hasFileExtension("xhtml")) {
```

Comment:
What happens when the file suffix is just .html?

Reply:
All of the RDF formats defined in the RDFFormat class which have file extension "html" also have file extension "xhtml". (And there is exactly one of them: the RDFA format--which is the one that is giving us problems.)

Reply:
ack, please ignore

Reply:
@lewismc I can also do the check via pattern matching on MIME types instead of file extensions, if you'd prefer. This way was just the easiest & quickest, and either way we'd get the same results.